### PR TITLE
Convert client script in settings pages to assets

### PR DESCRIPTION
**Pull request by jonatanschroeder**
* Move scripts from inline to asset files;
* Convert change ID button and associated form to a component;
* Move ID validation to an asset lib, using vanilla JS and browser native validation (proper use of `reportValidity()`);
* For URLs: moved clipboard button handling to asset lib;
* For URLs: moved QR code button handling to asset lib, using client-side QR rendering.
* Question settings: the page included validation for the question copy form, but it was impossible to fail the validation (check for blank target course, but target course is a select with no blank value), so I removed it.

### CODE REVIEW

File: apps/prairielearn/assets/scripts/lib/clipboardPopover.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import ClipboardJS from 'clipboard';
+
+import { onDocumentReady } from '@prairielearn/browser-utils';
+
+onDocumentReady(() => {
+  const clipboard = new ClipboardJS('.btn-copy');
```
[the remark this answers is not in the document]
Comment (jonatanschroeder) [reply apps/prairielearn/assets/scripts/lib/clipboardPopover.ts:6]: This is the same name used for a similar behaviour in PrairieTest (https://github.com/PrairieLearnInc/PrairieTest/blob/16591b61243556ea2fc2691c15a1299d1bb338e9/assets/scripts/behaviors/copy-button.ts#L4). Are you planning to change it there too?

File: apps/prairielearn/src/pages/instructorAssessmentSettings/instructorAssessmentSettings.html.ts [hunk 2]
```diff
@@ -186,15 +134,11 @@ export function InstructorAssessmentSettings({
                           type="button"
                           title="Student Link QR Code"
                           aria-label="Student Link QR Code"
-                          class="btn btn-sm btn-outline-secondary js-student-link-qrcode-button"
+                          class="btn btn-sm btn-outline-secondary js-qrcode-button"
+                          data-qr-code-content="${studentLink}"
```
[the remark this answers is not in the document]
Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/instructorAssessmentSettings/instructorAssessmentSettings.html.ts:138]: `data-content` does not support referring to SVG content directly for some reason. To get SVG to parse correctly I only managed to do it by passing an actual DOM element, which is why I had to resort to using `parseHTMLElement`. The old version used a hidden `div` with a direct reference to it, which is equivalent.

I'm ok with generating the SVG in server or client, I don't mind either. Generating it in the server requires the transfer of the SVG, generating it in the client requires the transfer of the QR library (possibly cached, tree-shaken and minified). I haven't checked how much is actually transferred in the client-side compiled asset version, but the library bundle has 18KB and the SVG for a QR code has ~90KB.

[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentSettings/instructorAssessmentSettings.html.ts:138]: Ahh, ok, I didn't realize that `data-content` didn't work with SVGs, though I guess that explains why the existing code doesn't use that either. Looks like the SVG library adds about ~6kb to the bundle, so it does seem like it's better to generate the SVG on the client. I'm happy to keep this.

File: apps/prairielearn/assets/scripts/lib/clipboardPopover.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/assets/scripts/lib/clipboardPopover.ts:6]: Gah, ok, fair, let's keep things consistent 😂

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**